flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/internal/converter/PostgresRowConverter.java

```diff
@@ -59,6 +63,25 @@ public JdbcDeserializationConverter createInternalConverter(LogicalType type) {
         }
     }
 
+    @Override
+    protected JdbcSerializationConverter createExternalConverter(LogicalType type) {
+        switch (type.getTypeRoot()) {
+            case CHAR:
+            case VARCHAR:
+                return (val, index, statement) -> {
+                    String valString = val.getString(index).toString();
+
+                    if (UUID_REGEX_PATTERN.matcher(valString).matches()) {
```

Comment:
@MartijnVisser I have strong doubts about this solution. This looks very error-prone and inefficient to me. This means we have to perform a regex matching for every string column. Also possibly UUID that are stored in a string column might be classified with a wrong data type.

Reply:
@bobby-richard After discussing this issue with @twalthr offline, we agreed that we indeed have to revert this change. The inefficiency and potential error cases are the driver for that. @twalthr has a different idea to resolve this, he'll create a fix for this soonish. Apologies for this situation.

Reply:
@MartijnVisser @twalthr I’m curious what the alternative solution is

Reply:
@bobby-richard Have a look at https://issues.apache.org/jira/browse/FLINK-29267

Reply:
Hello, I am curious if some work is done on  https://issues.apache.org/jira/browse/FLINK-29267 ?